lib/netext/resolver.go

```diff
@@ -162,21 +149,26 @@ func (r *resolver) selectOne(host string, ips []net.IP) net.IP {
 	return ip
 }
 
-func (r *resolver) applyPolicy(host string, ip4, ip6 []net.IP) (ip net.IP, ips []net.IP) {
-	switch r.policy {
+func (r *resolver) applyPolicy(ips []net.IP) (retIPs []net.IP) {
+	if r.policy == lib.DNSany {
+		return ips
+	}
+	ip4, ip6 := groupByVersion(ips)
+	switch r.policy { //nolint: exhaustive
```

Comment:
nit: Can you just have the exhaustive option and ... welll panic if we get to it? This way if we add another one it won't suddenly not be detected

Reply:
Fixed in the latest push, though I'm not panicking but just doing the fallback assignment.